network_acls.tf

```diff
@@ -39,40 +39,61 @@ locals {
       name        = "ibmflow-paas-outbound"
       action      = "allow"
       destination = "166.8.0.0/14"
-      source      = var.network_cidr != null ? var.network_cidr : "0.0.0.0/0"
+      source      = "0.0.0.0/0"
       direction   = "outbound"
       tcp         = null
       udp         = null
       icmp        = null
     }
   ]
 
-  vpc_inbound_rule = [
-    for address in data.ibm_is_vpc_address_prefixes.get_address_prefixes.address_prefixes :
-    {
-      name        = "ibmflow-allow-vpc-connectivity-inbound-${substr(address.id, -4, -1)}" # Providing unique rule names
-      action      = "allow"
-      source      = address.cidr
-      destination = var.network_cidr != null ? var.network_cidr : "0.0.0.0/0"
-      direction   = "inbound"
-      tcp         = null
-      udp         = null
-      icmp        = null
-    }
-  ]
-  vpc_outbound_rule = [
-    for address in data.ibm_is_vpc_address_prefixes.get_address_prefixes.address_prefixes :
-    {
-      name        = "ibmflow-allow-vpc-connectivity-outbound-${substr(address.id, -4, -1)}"
-      action      = "allow"
-      source      = var.network_cidr != null ? var.network_cidr : "0.0.0.0/0"
-      destination = address.cidr
-      direction   = "outbound"
-      tcp         = null
-      udp         = null
-      icmp        = null
-    }
-  ]
+  ibm_cloud_internal_rules = flatten([
+    for rules in local.internal_rules : [
+      for index, cidrs in var.network_cidrs != null ? var.network_cidrs : ["0.0.0.0/0"] : {
```

Comment:
why do we override all properties defined already in `internal_rules`? why don't we add/override only `source`?

Reply:
updated the logic.